frontend/src/components/ProjectDeploymentConfigPanel.vue

```diff
@@ -198,40 +207,47 @@ export default defineComponent({
       });
     });
 
-    watchEffect(() => {
-      const dep = store.getters["deployment/deploymentConfigByProjectId"](
+    const resetStates = async () => {
+      await nextTick(); // Waiting for all watchers done
+      state.dirty = false;
```

Comment:
The existing pattern is not using the dirty flag. Instead, the code keeps an original deployment and compares that with the effective deployment.

See https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/frontend/src/components/InstanceForm.vue?L303